### PR TITLE
Swarm integration

### PR DESCRIPTION
**Pull request by MaiaVictor**
Initial Swarm release. I migrated most of the complicate Swarm logic to its own repository, [Swarm-js](https://github.com/maiavictor/swarm-js). This allows Swarm to be implemented and tested independently, and makes things much simpler on Mist's side, which now just needs thin wrappers to that library. This is an ongoing work and there is still a lot to do.

### CODE REVIEW

**Comment by alexvandesande** [comment]
When you try to access a bad hash it says "Manifest not Found". Can we intercept that error and show a mist 404 error page?

**Comment by frozeman** [comment]
Two things:
- The wallet can not be served via swarm (yet), as it would change the local storage location and this removes all peoples wallets. So until we don't have a official ENS domain we can host it on swarm.
- The upload directory is not there yet in the swarm > upload menu entry

**Comment by MaiaVictor** [comment]
@frozeman indeed. The `upload` entry now accepts both files and directories. It will also detect an `index.html` and set it as `defaultFile`, so that you can upload an web app by merely choosing a directory with an `index.html`.

@alexvandesande I think so, will investigate this soon, as well as showing a directory tree when it is a manifest without a default HTML.

**Comment by frozeman** [comment]
Get some errors when resolving urls:
```
[2017-06-11 21:22:20.778] [INFO] Sockets/2 - Disconnecting...
Unhandled rejection TypeError: Cannot read property 'removeAllListeners' of undefined
    at _disconnectPromise.Q (/Users/xxxxx/Sites/_ethereum/mist/modules/sockets/base.js:131:29)
    at Promise.cancellationExecute [as _execute] (/Users/xxxxx/Sites/_ethereum/mist/node_modules/bluebird/js/release/debuggability.js:321:9)
    at Promise._resolveFromExecutor (/Users/xxxxx/Sites/_ethereum/mist/node_modules/bluebird/js/release/promise.js:481:18)
    at new Promise (/Users/xxxxx/Sites/_ethereum/mist/node_modules/bluebird/js/release/promise.js:77:14)
    at Web3IpcSocket.disconnect (/Users/xxxxx/Sites/_ethereum/mist/modules/sockets/base.js:125:39)
    at Web3IpcSocket.destroy (/Users/xxxxx/Sites/_ethereum/mist/modules/sockets/base.js:173:21)
    at IpcProviderBackend._destroyConnection (/Users/xxxxx/Sites/_ethereum/mist/modules/ipc/ipcProviderBackend.js:196:47)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:191:7)
    at WebContents.<anonymous> (/Users/xxxxx/.config/yarn/global/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/browser/api/web-contents.js:240:37)
    at emitTwo (events.js:106:13)
    at WebContents.emit (events.js:191:7)
```

**Comment by destinatis** [comment]
Love it.

File: interface/client/lib/helpers/helperFunctions.js [hunk 1]
```diff
@@ -79,8 +79,11 @@ Format Urls, e.g add a default protocol if on is missing.
 @param {String} url
 **/
 Helpers.formatUrl = function (url) {
-    // add http:// if no protocol is present
-    if (url && url.indexOf('://') === -1) {
+    if (url && url.length === 64 && !!url.match(/^[0-9a-f]+$/)) {
+        // if the url looks like a hash, add bzz
```
Comment (nicksavers) [inline interface/client/lib/helpers/helperFunctions.js]: Maybe also include a check if the domain is `.eth`

**Pull request closed without merging**